.buildkite/scripts/test-with-integrations.sh

```diff
@@ -0,0 +1,188 @@
+#!/bin/bash
+
+set -euo pipefail
+
+WORKSPACE="$(pwd)"
+
+TMP_FOLDER_TEMPLATE_BASE="tmp.${GITHUB_PR_BASE_REPO}"
+TMP_FOLDER_TEMPLATE="${TMP_FOLDER_TEMPLATE_BASE}.XXXXXXXXX"
+
+cleanup() {
+    echo "Deleting temporal files..."
+    cd ${WORKSPACE}
+    rm -rf "${TMP_FOLDER_TEMPLATE_BASE}.*"
+    echo "Done."
+}
+
+trap cleanup EXIT
+source .buildkite/scripts/install_deps.sh
+
+add_bin_path
+
+echo "--- install gh cli"
+with_github_cli
+
+echo "--- install jq"
+with_jq
+
+
+INTEGRATIONS_SOURCE_BRANCH=main
+INTEGRATIONS_GITHUB_OWNER=elastic
+INTEGRATIONS_GITHUB_REPO_NAME=integrations
+INTEGRATIONS_PR_BRANCH="test-${GITHUB_PR_BASE_REPO}-pr-${BUILDKITE_PULL_REQUEST}"
+INTEGRATIONS_PR_TITLE="Test ${GITHUB_PR_BASE_REPO} - DO NOT MERGE"
+VERSION_DEP=""
+
+get_pr_number() {
+    # requires GITHUB_TOKEN
+    local branch="$1"
+    gh pr list -H "${branch}" --json number | jq -r '.[]|.number'
+}
+
+get_integrations_pr_link() {
+    local pr_number=$1
+    echo "https://github.com/elastic/integrations/pull/${pr_number}"
+}
+
+get_source_pr_link() {
+    echo "https://github.com/${GITHUB_PR_BASE_OWNER}/${GITHUB_PR_BASE_REPO}/pull/${BUILDKITE_PULL_REQUEST}"
+}
+
+get_source_commit_link() {
+    echo "https://github.com/${GITHUB_PR_BASE_OWNER}/${GITHUB_PR_BASE_REPO}/commit/${GITHUB_PR_HEAD_SHA}"
+}
+
+set_git_config() {
+    git config user.name "${GITHUB_USERNAME_SECRET}"
+    git config user.email "${GITHUB_EMAIL_SECRET}"
+}
+
+git_push_with_auth() {
+    local owner="$1"
+    local repository="$2"
+    local branch="$3"
+
+    retry 3 git push https://${GITHUB_USERNAME_SECRET}:${GITHUB_TOKEN}@github.com/${owner}/${repository}.git "${branch}"
+}
+
+clone_repository() {
+    local target="$1"
+    retry 5 git clone https://github.com/elastic/integrations ${target}
+}
+
+create_integrations_pull_request() {
+    # requires GITHUB_TOKEN
+    local temp_path=$(mktemp -d -p ${WORKSPACE} -t ${TMP_FOLDER_TEMPLATE})
+    echo "Creating Pull Request"
+    message="Update ${GITHUB_PR_BASE_REPO} reference to $(get_source_commit_link).\nAutomated by [Buildkite build](${BUILDKITE_BUILD_URL})\n\nRelates: $(get_source_pr_link)"
+    echo -e $message > ${temp_path}/body-pr.txt
+    retry 3 \
+        gh pr create \
+        --title "${INTEGRATIONS_PR_TITLE}" \
+        --body-file ${temp_path}/body-pr.txt \
+        --draft \
+        --base ${INTEGRATIONS_SOURCE_BRANCH} \
+        --head ${INTEGRATIONS_PR_BRANCH} \
+        --assignee ${GITHUB_PR_HEAD_USER}
+}
```

Comment:
Instead of creating a pull request, would it be possible to trigger an integrations build, parameterized with the version of elastic-package, and report the status on the elastic-package PR?

Reply:
That would be adding more logic to the integrations job.
An advantage of creating that pull request is that everything is tracked in one place, and it can be seen the evolution.